src/views/CardStack/CardStackTransitioner.js

```diff
@@ -53,11 +54,26 @@ class CardStackTransitioner extends React.Component {
     return transitionSpec;
   };
 
+  _getMode() {
+    const { navigation, mode, router } = this.props;
+    // If there's a route, get its screen options
+    if (navigation.state.routes && navigation.state.routes[1]) {
+      screenOptions = router.getScreenOptions(
+        addNavigationHelpers({
+          state: navigation.state.routes[1],
```

Comment:
It looks like we will only check the options of routes[1], which is the second screen in the stack. What if the third route is configured to be a modal? Can you try this out in the navigation playground and possibly post a video of what it looks like?